src/Sylius/Bundle/AdminBundle/Console/Command/ListAdminUsersCommand.php

```diff
@@ -0,0 +1,127 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\AdminBundle\Console\Command;
+
+use Sylius\Component\Core\Model\AdminUserInterface;
+use Sylius\Component\User\Repository\UserRepositoryInterface;
+use Symfony\Component\Console\Attribute\AsCommand;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Helper\Table;
+use Symfony\Component\Console\Helper\TableStyle;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Style\SymfonyStyle;
+
+#[AsCommand(
+    name: 'sylius:admin-user:list',
+    description: 'Displays all admin users with information in a list. Optionally, also one user can be displayed if the username is specified as an option.',
+)]
+final class ListAdminUsersCommand extends Command
+{
+    protected SymfonyStyle $io;
+
+    /** @param UserRepositoryInterface<AdminUserInterface> $userRepository */
+    public function __construct(
+        private readonly UserRepositoryInterface $userRepository,
+    ) {
+        parent::__construct();
+    }
+
+    protected function initialize(InputInterface $input, OutputInterface $output): void
+    {
+        $this->io = new SymfonyStyle($input, $output);
+    }
+
+    protected function configure(): void
+    {
+        $this->addOption(
+            'search',
+            's',
+            InputOption::VALUE_REQUIRED,
+            'Search users by email, username, first name or last name (partial match)',
+        );
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output): int
+    {
+        $this->io->title('List available admin users');
+        $searchTerm = $input->getOption('search');
+        if ($searchTerm === null || $searchTerm === '') {
+            $this->renderUsersTable($output, $this->userRepository->findAll());
+
+            return Command::SUCCESS;
+        }
+        $users = $this->searchUsers($searchTerm);
+        if (empty($users)) {
+            $this->io->warning(sprintf('No users found matching "%s"', $searchTerm));
+
+            return Command::SUCCESS;
+        }
+        $this->io->success(sprintf('Found %d user(s) matching "%s"', count($users), $searchTerm));
+        $this->renderUsersTable($output, $users);
+
+        return Command::SUCCESS;
+    }
```

Comment:
```suggestion
        $this->io->title('List available admin users');
        
        $searchTerm = $input->getOption('search');
        if ($searchTerm === null || $searchTerm === '') {
            $this->renderUsersTable($output, $this->userRepository->findAll());

            return Command::SUCCESS;
        }
        
        $users = $this->searchUsers($searchTerm);
        if (empty($users)) {
            $this->io->warning(sprintf('No users found matching "%s"', $searchTerm));

            return Command::SUCCESS;
        }
        
        $this->io->success(sprintf('Found %d user(s) matching "%s"', count($users), $searchTerm));
        $this->renderUsersTable($output, $users);

        return Command::SUCCESS;
    }
```

Let's add whitelines for readability.